csrc/cpu/comm/ccl.cpp

```diff
@@ -263,11 +263,6 @@ void inference_all_reduce(torch::Tensor& data, py::object op, bool async_op)
     auto start = std::chrono::system_clock::now();
 #endif
 
-    static py::object ReduceOp = py::module_::import("deepspeed.comm").attr("ReduceOp");
```

Comment:
I think it is confusing to drop this assert since `inference_all_reduce` only works for sum even though it is not clear from the name. Users can easily assume that this function is similar to more common ones like nccl allreduce that supports other user-defined reduction ops like MAX. I see two possible solutions:
1. Restore this assert so that users are notified of unsupported ops, and developers can eventually support more ops without changing API.
2. Rename function to something like `inference_all_reduce_sum` to better reflect the intention.

Reply:
Hi @tjruwase I think leave some space for developer to extend might be a better option.  Let me add the assertion back.